browser/extensions/api/playlist/playlist_event_router_factory.cc

```diff
@@ -0,0 +1,108 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/extensions/api/playlist/playlist_event_router_factory.h"
+
+#include <memory>
+#include <string>
+#include <utility>
+
+#include "base/scoped_observer.h"
+#include "brave/browser/playlist/playlist_service_factory.h"
+#include "brave/common/extensions/api/playlist.h"
+#include "brave/components/playlist/playlist_service.h"
+#include "brave/components/playlist/playlist_service_observer.h"
+#include "components/keyed_service/content/browser_context_dependency_manager.h"
+#include "components/keyed_service/core/keyed_service.h"
+#include "extensions/browser/event_router.h"
+
+namespace OnPlaylistItemStatusChanged =
+    extensions::api::playlist::OnPlaylistItemStatusChanged;
+
+namespace playlist {
+
+namespace {
+
+PlaylistService* GetPlaylistService(content::BrowserContext* context) {
+  return PlaylistServiceFactory::GetInstance()->GetForBrowserContext(context);
+}
+
+}  // namespace
+
+class PlaylistEventRouterFactory::PlaylistEventRouter
```

Comment:
why are we layering PlaylistEventRouter on top of extensions::EventRouter? It would make sense if `PlaylistEventRouter` had to also handle events from something other than extensions, but I don't understand it here